src/ripple/rpc/handlers/PayChanClaim.cpp

```diff
@@ -36,16 +36,23 @@ namespace ripple {
 
 // {
 //   secret_key: <signing_secret_key>
+//   key_type: optional; either ed25519 or secp256k1 (default to secp256k1)
 //   channel_id: 256-bit channel id
 //   drops: 64-bit uint (as string)
 // }
 Json::Value doChannelAuthorize (RPC::Context& context)
 {
     auto const& params (context.params);
-    for (auto const& p : {jss::secret, jss::channel_id, jss::amount})
+    for (auto const& p : {jss::channel_id, jss::amount})
         if (!params.isMember (p))
             return RPC::missing_field_error (p);
 
+    // Compatibility if a key type isn't specified. If it is, the
+    // keypairForSignature code will validate parameters and return
+    // the appropriate error.
+    if (!params.isMember(jss::key_type) && !params.isMember(jss::secret))
+        return RPC::missing_field_error (jss::secret);
```

Comment:
FWIW, codecov indicates that the new unit tests in PayChan_test.cpp don't exercise this line or line 48 for that matter.  Also there are other lines in the vicinity that are not exercised by any of the unit tests.

If you would like, you can get coverage for all the lines in this method by cherry-picking this commit: https://github.com/scottschurr/rippled/commit/d9b2cc91ea1016f444fbb0d2f3896dfc39c8c396.  If you do choose to cherry-pick it, feel free to adapt it as you see fit.